### PR TITLE
Add test cases for source-build

### PR DESCRIPTION
**Pull request by tkdchen**
[STONEBLD-1874](https://issues.redhat.com//browse/STONEBLD-1874)

Relates to the changes of resolving parent images made in build-tasks-dockerfiles/source-container-build.

Requires https://github.com/redhat-appstudio/build-tasks-dockerfiles/pull/33

### CODE REVIEW

File: .tekton/tasks/e2e-test.yaml [hunk 1]
```diff
@@ -31,7 +31,7 @@ spec:
       - name: APP_SUFFIX
         value: "$(params.app_suffix)"
       - name: COMPONENT_REPO_URLS
-        value: "https://github.com/redhat-appstudio-qe/devfile-sample-python-basic,https://github.com/redhat-appstudio-qe/retrodep,https://github.com/cachito-testing/pip-e2e-test,https://github.com/redhat-appstudio-qe/fbc-sample-repo,https://github.com/redhat-appstudio-qe/nodejs-no-dockerfile,https://github.com/redhat-appstudio-qe/maven-hello-world"
+        value: "https://github.com/redhat-appstudio-qe/devfile-sample-python-basic,https://github.com/redhat-appstudio-qe/retrodep,https://github.com/cachito-testing/pip-e2e-test,https://github.com/redhat-appstudio-qe/fbc-sample-repo,https://github.com/redhat-appstudio-qe/nodejs-no-dockerfile,https://github.com/redhat-appstudio-qe/maven-hello-world,https://github.com/cqi-stonesoup-test/e2e-tests-parent-image-with-both-tag-digest,https://github.com/cqi-stonesoup-test/e2e-tests-parent-image-with-digest-only,https://github.com/cqi-stonesoup-test/e2e-tests-use-latest-parent-image"
```
Comment (tkdchen) [inline .tekton/tasks/e2e-test.yaml:34]: @tisutisu Should all the repositories added here be under `redhat-appstudio-qe`?

Comment (tisutisu) [reply .tekton/tasks/e2e-test.yaml:34]: Yes, I think it would be good to them under `redhat-appstudio-qe`

Comment (tkdchen) [reply .tekton/tasks/e2e-test.yaml]: Done

**Review by chmeliik — COMMENTED** [review]
> Relates to the changes of resolving parent images made in
build-tasks-dockerfiles/source-container-build.

By adding the component urls, we only check that the builds succeed right? This doesn't really test the parent image resolution, does it?

**Comment by chmeliik** [comment]
> There is already a story in the EPIC for updating the e2e tests.

Ah, ok :+1:

**Review by chmeliik — APPROVED** [review]

**Review by tisutisu — APPROVED** [review]
LGTM

**Pull request merged**